### PR TITLE
Gradle 4.10 release notes/features

### PR DESCRIPTION
**Pull request by eriwen**
### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -1,3 +1,29 @@
+The Gradle team is pleased to announce Gradle 4.10. This is a big release.
+
+First and foremost, this release of Gradle features an improved [incremental Java compiler, now enabled by default](#incremental-java-compilation-by-default).
+This will result in significantly reduced Java compilation time in subsequent builds when outputs are not up-to-date or resolved from the build cache.
+
+Chances are caches in those `.gradle/` directories have accumulated a few (or a few dozen) gigabytes over time.
+If so, you'll be relieved to know that Gradle will now [periodically clean up unused `/caches`](#periodic-cache-cleanup) under `GRADLE_USER_HOME` and project root directories. 
+
+A moment you have anticipated is nearly here, as the [Kotlin DSL reaches version 1.0 RC1](TODO) with this release of Gradle.
+TODO @eskatos: details about configuration avoidance and other changes.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md]: I'd add some encouragement here to give it a try and provide feedback.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+
+You can now use [SNAPSHOT plugin versions with the `plugins {}`](#use-snapshot-plugin-versions-with-the-plugins-{}-block) and `pluginManagement {}` blocks.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md]: Shall we elaborate on why you may want to do that?

**Comment by pioterj** [comment]
LGTM. I left few comments.

File: subprojects/docs/src/docs/release/release-features.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+ - Incremental Java compilation by default
+ - Automatic Gradle caches/ cleanup
```
Comment (marcphilipp) [inline subprojects/docs/src/docs/release/release-features.txt]: `- Periodic cache cleanup`?

Comment (eriwen) [reply subprojects/docs/src/docs/release/release-features.txt]: Perhaps "periodic" is better, but I'm also hoping to make it a little more specific by saying "Gradle" and "caches/" dir (knowing full well it's caches\ on Windows)

Comment (eriwen) [reply subprojects/docs/src/docs/release/release-features.txt]: "Periodic Gradle caches cleanup"?

Comment (marcphilipp) [reply subprojects/docs/src/docs/release/release-features.txt]: It's not just the `caches` dir, it's also `wrapper/dists`. 😉 

I'll leave it up to you but I found the trailing slash confusing. Looked like a typo to me.

**Pull request merged**